hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java

```diff
@@ -464,4 +487,138 @@ private static Account getEffectiveOwnerAccount(
+        final var adjustedBytes = getNewBytes(body.cryptoApproveAllowanceOrThrow(), account);
+        return feeContext
+                .feeCalculator(SubType.DEFAULT)
+                .addBytesPerTransaction(bytesUsedInTxn(op))
+                .addRamByteSeconds(adjustedBytes > 0 ? (adjustedBytes * lifeTime) : 0)
+                .calculate();
+    }
+
+    /**
+     * Gets total bytes used in transaction
+     * @param op the crypto approve allowance transaction body
+     * @return the total bytes used in transaction
+     */
+    private int bytesUsedInTxn(final CryptoApproveAllowanceTransactionBody op) {
+        return op.cryptoAllowancesOrElse(emptyList()).size() * CRYPTO_ALLOWANCE_SIZE
+                + op.tokenAllowancesOrElse(emptyList()).size() * TOKEN_ALLOWANCE_SIZE
+                + op.nftAllowancesOrElse(emptyList()).size() * NFT_ALLOWANCE_SIZE
+                + countSerials(op.nftAllowancesOrElse(emptyList())) * LONG_SIZE;
+    }
+
+    /**
+     * Gets the new bytes that will be added to state from the transaction, if it is successful compared to
+     * what is already present in state
+     * @param op the crypto approve allowance transaction body
+     * @param account the account existing in state
+     * @return the new bytes that will be added to state
+     */
+    private long getNewBytes(final CryptoApproveAllowanceTransactionBody op, final Account account) {
+        final long newCryptoKeys = getChangedCryptoKeys(
+                op.cryptoAllowancesOrElse(emptyList()),
+                account == null ? emptyList() : account.cryptoAllowancesOrElse(emptyList()));
+        final long newTokenKeys = getChangedTokenKeys(
+                op.tokenAllowancesOrElse(emptyList()),
+                account == null ? emptyList() : account.tokenAllowancesOrElse(emptyList()));
+        final long newApproveForAllNfts = getChangedNftKeys(
+                op.nftAllowancesOrElse(emptyList()),
+                account == null ? emptyList() : account.approveForAllNftAllowancesOrElse(emptyList()));
+
+        return newCryptoKeys * CRYPTO_ALLOWANCE_SIZE
+                + newTokenKeys * TOKEN_ALLOWANCE_SIZE
+                + newApproveForAllNfts * NFT_ALLOWANCE_SIZE;
+    }
+
+    /**
+     * Gets the number of new crypto allowances that will be added to state from the transaction.
+     * @param newAllowances the list of new crypto allowances
+     * @param existingAllowances the list of existing crypto allowances
+     * @return the number of new crypto allowances that will be added to state from the transaction
+     */
+    private int getChangedCryptoKeys(
+            final List<CryptoAllowance> newAllowances, final List<AccountCryptoAllowance> existingAllowances) {
+        int counter = 0;
+        for (var key : newAllowances) {
+            for (int i = 0; i < existingAllowances.size(); i++) {
+                if (existingAllowances.get(i).spenderId().equals(key.spender())) {
+                    break;
```

Comment:
Why does this `break` instead of `continue`-ing?

Reply:
Once we find the allowance exists, no need to look for more.

Reply:
Once we find allowance exists, no need to count again. But I did change the logic.

Reply:
Ahhh got it, thanks for clarifying!